jobs/build-node-image.Jenkinsfile

```diff
@@ -0,0 +1,183 @@
+def gitref, commit, shortcommit
+
+node {
+    checkout scm
+    // these are script global vars
+    pipeutils = load("utils.groovy")
+    pipecfg = pipeutils.load_pipecfg()
+}
+
+properties([
+    pipelineTriggers([cron('H H * * *')]),
+    parameters([
+      choice(name: 'RELEASE',
+             choices: pipeutils.get_streams_choices(pipecfg, true),
+             description: 'CoreOS stream to build'),
+      string(name: 'FROM',
+             description: 'Image FROM',
+             defaultValue: '',
+             trim: true),
+      string(name: 'ARCHES',
+             description: 'Space-separated list of target architectures',
+             defaultValue: "x86_64 aarch64 ppc64le s390x",
+             trim: true),
+      string(name: 'COREOS_ASSEMBLER_IMAGE',
+             description: 'Override the coreos-assembler image to use',
+             defaultValue: "quay.io/coreos-assembler/coreos-assembler:latest",
+             trim: true),
+    ]),
+    buildDiscarder(logRotator(
+        numToKeepStr: '100',
+        artifactNumToKeepStr: '100'
+    )),
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+])
+
+// Note the supermin VM just uses 2G. The really hungry part is xz, which
+// without lots of memory takes lots of time. For now we just hardcode these
+// here; we can look into making them configurable through the template if
+// developers really need to tweak them.
+// XXX bump an extra 2G (to 10.5) because of an error we are seeing in
+// testiso: https://github.com/coreos/fedora-coreos-tracker/issues/1339
+def cosa_memory_request_mb = 4.5 * 1024 as Integer
+
+// Now that we've established the memory constraint based on xz above, derive
+// kola parallelism from that. We leave 512M for overhead and VMs are at most
+// 1.5G each (in general; root reprovisioning tests require 4G which is partly
+// why we run them separately).
+// XXX: https://github.com/coreos/coreos-assembler/issues/3118 will make this
+// cleaner
+def ncpus = ((cosa_memory_request_mb - 512) / 1536) as Integer
+
+// runtime parameter always wins
+def cosa_img = params.COREOS_ASSEMBLER_IMAGE
+
+def stream_info = pipecfg.node_builds.ocp_release[params.RELEASE]
+def src_config_ref = stream_info.source_config.ref
+def src_config_url = stream_info.source_config.url
+
+node {
+
+    change = checkout(
+        changelog: true,
+        poll: false,
+        scm: [
+            $class: 'GitSCM',
+            branches: [[name: "origin/${src_config_ref}"]],
+            userRemoteConfigs: [[url: src_config_url]],
+            extensions: [[$class: 'CloneOption',
+                          noTags: true,
+                          reference: '',
+                          shallow: true]]
+        ]
+    )
+
+    def output = shwrapCapture("git rev-parse HEAD")
+    commit = output.substring(0,40)
+    shortcommit = commit.substring(0,7)
+}
```

Comment:
Is it even worth cloning the repo here? I think instead we could just `git ls-remote` from within the cosa pod.